pkg/api/types.go

```diff
@@ -195,6 +195,119 @@ type VolumeSource struct {
 	NFS *NFSVolumeSource `json:"nfs"`
 }
 
+// Similar to VolumeSource but meant for the administrator who creates PVs.
+// Exactly one of its members must be set.
+type PersistentVolumeSource struct {
+	// GCEPersistentDisk represents a GCE Disk resource that is attached to a
+	// kubelet's host machine and then exposed to the pod.
+	GCEPersistentDisk *GCEPersistentDiskVolumeSource `json:"gcePersistentDisk"`
+	// HostPath represents a directory on the host.
+	// This is useful for development and testing only.
+	// on-host storage is not supported in any way
+	HostPath *HostPathVolumeSource `json:"hostPath"`
+}
+
+type PersistentVolume struct {
+	TypeMeta   `json:",inline"`
+	ObjectMeta `json:"metadata,omitempty"`
+
+	//Spec defines a persistent volume owned by the cluster
```

Comment:
@quinton-hoole Allow me to split this hair for you :)

You are correct that nodes are system infrastructure.  They represent resources in the infrastructure.  A PersistentVolume is the same thing.  It is a provisioned resource.

A PersistentVolumeClaim is the user application domain you refer to.  Users request a slice of resources by creating a claim (as they request a slice of computing resource by creating a pod).  Users create, use, and delete claims.

We are introducing another volume plugin that allows the user to use their claim as their volume.